octoprint_curalegacy/profile.py

```diff
@@ -756,7 +756,10 @@ def get_gcode(self, key, extruder_count=1):
 		else:
 			contents = self.get_gcode_template(key, extruder_count=extruder_count)
 
-		return unicode(prefix + re.sub("(.)\{([^\}]*)\}", self.replaceTagMatch, contents).rstrip() + '\n' + postfix).strip().encode('utf-8') + b'\n'
+		gcode = unicode(prefix + re.sub("(.)\{([^\}]*)\}", self.replaceTagMatch, contents).rstrip() + '\n' + postfix).strip().encode('utf-8') + b'\n'
```

Comment:
That sounds... weird. Returning unicode by this method in general should not cause issues. And this code snippet here will behave differently on Py2 vs 3, whereas the goal should be identical behaviour wherever possible.

Can you elaborate on what exactly you saw in your tests that made you make this change?

Reply:
Yes.  Sarge throws under Python 3 because it gets an array of bytes instead of unicode due to that + b'\n' at the end.  Now it's been long enough I don't know if I tried to just have it remain unicode for Python 2 or not, but I have a vague recollection that Sarge under python 2 throws if it gets unicode.  In any case, with this proposed change I had convinced myself that sarge wanted different behavior under python 2 than python 3.  I'll have to repro again to make sure though.

Reply:
Nope. Apparently not.  Looks like I was assuming rather than testing and trying too hard not to change the python2 code path in a legacy component.